### PR TITLE
Adds job playtime restrictions.

### PR DESCRIPTION
**Pull request by DominikPanic**
:cl: LaKiller8
tweak: Job restrictions have been enabled, with 3 hours for CPT, Synth, SL, FCDR, one hour for head departments and the rest of command, and one hour for the xeno queen
/:cl:

### CODE REVIEW

File: code/game/jobs/job/job_exp.dm [hunk 1]
```diff
@@ -240,4 +240,18 @@ GLOBAL_PROTECT(exp_to_update)
 			"ckey" = "'[sanitizeSQL(ckey)]'",
 			"minutes" = jvalue)))
 		prefs.exp[jtype] += jvalue
-	addtimer(CALLBACK(GLOBAL_PROC, /proc/update_exp_db), 20, TIMER_OVERRIDE|TIMER_UNIQUE)
\ No newline at end of file
+	addtimer(CALLBACK(GLOBAL_PROC, /proc/update_exp_db), 20, TIMER_OVERRIDE|TIMER_UNIQUE)
+
+
+/proc/queen_age_check(client/C)
+	if(!istype(C) || !C.prefs?.exp)
```
Comment (Rohesie) [inline code/game/jobs/job/job_exp.dm]: This proc is being called by passing a client. The sanity check shouldn't be necessary.

File: code/game/jobs/job/job_exp.dm [hunk 2]
```diff
@@ -240,4 +240,18 @@ GLOBAL_PROTECT(exp_to_update)
...
+		return FALSE
+	if(!CONFIG_GET(flag/use_exp_tracking))
+		return FALSE
+	if(CONFIG_GET(flag/use_exp_restrictions_admin_bypass) && check_other_rights(C, R_ADMIN, FALSE))
+		return FALSE
+	var/my_exp = C.prefs.exp[ROLE_XENOMORPH]
+	if(my_exp >= 60)
+		return FALSE
+	else
+		return (60 - my_exp)
```
Comment (Rohesie) [inline code/game/jobs/job/job_exp.dm]:
```suggestion
		return TRUE
```
You have already established `my_exp` is lower than 60.

File: code/game/jobs/job/job_exp.dm [hunk 3]
```diff
@@ -240,4 +240,18 @@ GLOBAL_PROTECT(exp_to_update)
...
+	if(!C.prefs?.exp)
+		return FALSE
+	if(!CONFIG_GET(flag/use_exp_tracking))
+		return FALSE
+	if(CONFIG_GET(flag/use_exp_restrictions_admin_bypass) && check_other_rights(C, R_ADMIN, FALSE))
+		return FALSE
+	var/my_exp = C.prefs.exp[ROLE_XENOMORPH]
+	if(my_exp >= 60)
+		return FALSE
+	else
```
Comment (Rohesie) [inline code/game/jobs/job/job_exp.dm]: I don't think it's much of an issue, but Ghommie and Spooky have been pointing at needless `else`.

File: code/game/jobs/job/shipside.dm [hunk 4]
```diff
@@ -101,6 +105,8 @@ Make the TGMC proud!"})
 	skills_type = /datum/skills/SO
 	display_order = JOB_DISPLAY_ORDER_STAFF_OFFICER
 	outfit = /datum/outfit/job/command/intelligenceofficer
+	exp_requirements = 60
```
Comment (Rohesie) [inline code/game/jobs/job/shipside.dm]: Make this a XP_REQ_INTERMEDIATE define.

File: code/game/jobs/job/shipside.dm [hunk 5]
```diff
@@ -60,6 +62,8 @@ Godspeed, captain! And remember, you are not above the law."})
 	minimal_access = ALL_MARINE_ACCESS
 	display_order = JOB_DISPLAY_ORDER_EXECUTIVE_OFFICER
 	outfit = /datum/outfit/job/command/fieldcommander
+	exp_requirements = 180
```
Comment (Rohesie) [inline code/game/jobs/job/shipside.dm]: This can be a XP_REQ_EXPERIENCED

File: code/game/jobs/job/job_exp.dm [hunk 6]
```diff
@@ -240,4 +240,18 @@ GLOBAL_PROTECT(exp_to_update)
...
+
+/proc/queen_age_check(client/C)
+	if(!C.prefs?.exp)
+		return FALSE
+	if(!CONFIG_GET(flag/use_exp_tracking))
+		return FALSE
+	if(CONFIG_GET(flag/use_exp_restrictions_admin_bypass) && check_other_rights(C, R_ADMIN, FALSE))
+		return FALSE
+	var/my_exp = C.prefs.exp[ROLE_XENOMORPH]
+	if(my_exp >= 60)
```
Comment (spookydonut) [inline code/game/jobs/job/job_exp.dm]: just `return my_exp < 60` instead of using an if

**Pull request merged**